### Questions

### Question 1

- Le fichier cwurData : un classement des universités par années et par Pays, il y'a plusieurs classements visiblement à l'intérieur

- Le fichier educational_attainment_supplementary_data donne par Pays et tranche d'âge des notes selon les années

- Le fichier school and country table associe des écoles et des Pays

- Le fichier Sanghai data classe des universités par année

- Le fichier timesData classe des universités avec pas mal de critères (différentes notes, élèves étrangers ...) 

- Le fichier education_expenditure_supplementary_data classe les Pays par niveau d'éducation (école primaire, ...)


In [1]:
import numpy as np
import pandas as pd
import os 

In [126]:
data1 = pd.read_csv("./kaggle_université/cwurData.csv", encoding='latin-1', sep=",")
df1 = pd.DataFrame(data1)
df1.head()

world_rank                            institution         country  \
0           1                     Harvard University             USA   
1           2  Massachusetts Institute of Technology             USA   
2           3                    Stanford University             USA   
3           4                University of Cambridge  United Kingdom   
4           5     California Institute of Technology             USA   

   national_rank  quality_of_education  alumni_employment  quality_of_faculty  \
0              1                     7                  9                   1   
1              2                     9                 17                   3   
2              3                    17                 11                   5   
3              1                    10                 24                   4   
4              4                     2                 29                   7   

   publications  influence  citations  broad_impact  patents   score  year  
0             1          1          1           NaN        5  100.00  2012  
1            12          4          4           NaN        1   91.67  2012  
2             4          2          2           NaN       15   89.50  2012  
3            16         16         11           NaN       50   86.17  2012  
4            37         22         22           NaN       18   85.21  2012

In [10]:
data2 = pd.read_csv("./kaggle_université/educational_attainment_supplementary_data.csv", encoding='latin-1', sep=",")
df2 = pd.DataFrame(data2)
df2.head()

country_name                                        series_name  1985  1986  \
0  Afghanistan  Barro-Lee: Average years of primary schooling,...  0.33   NaN   
1  Afghanistan  Barro-Lee: Average years of primary schooling,...  1.03   NaN   
2  Afghanistan  Barro-Lee: Average years of primary schooling,...  0.83   NaN   
3  Afghanistan  Barro-Lee: Average years of primary schooling,...  2.34   NaN   
4  Afghanistan  Barro-Lee: Average years of primary schooling,...  0.54   NaN   

   1987  1990  1991  1992  1993  1995  ...  2005  2006  2007  2008  2009  \
0   NaN  0.44   NaN   NaN   NaN  0.57  ...  0.86   NaN   NaN   NaN   NaN   
1   NaN  1.26   NaN   NaN   NaN  1.54  ...  2.18   NaN   NaN   NaN   NaN   
2   NaN  0.95   NaN   NaN   NaN  1.26  ...  1.01   NaN   NaN   NaN   NaN   
3   NaN  2.22   NaN   NaN   NaN  2.37  ...  2.26   NaN   NaN   NaN   NaN   
4   NaN  0.92   NaN   NaN   NaN  0.94  ...  2.00   NaN   NaN   NaN   NaN   

   2010  2011  2012  2013  2015  
0  1.27   NaN   NaN   NaN   NaN  
1  2.64   NaN   NaN   NaN   NaN  
2  2.45   NaN   NaN   NaN   NaN  
3  3.55   NaN   NaN   NaN   NaN  
4  1.29   NaN   NaN   NaN   NaN  

[5 rows x 29 columns]

In [11]:
data3 = pd.read_csv("./kaggle_université/school_and_country_table.csv", encoding='latin-1', sep=",")
df3 = pd.DataFrame(data3)
df3.head()

school_name                   country
0                     Harvard University  United States of America
1     California Institute of Technology  United States of America
2  Massachusetts Institute of Technology  United States of America
3                    Stanford University  United States of America
4                   Princeton University  United States of America

In [13]:
data4 = pd.read_csv("./kaggle_université/shanghaiData.csv", encoding='latin-1', sep=",")
df4 = pd.DataFrame(data4)
df4.head()

world_rank                              university_name national_rank  \
0          1                           Harvard University             1   
1          2                      University of Cambridge             1   
2          3                          Stanford University             2   
3          4           University of California, Berkeley             3   
4          5  Massachusetts Institute of Technology (MIT)             4   

   total_score  alumni  award   hici     ns    pub   pcp  year  
0        100.0   100.0  100.0  100.0  100.0  100.0  72.4  2005  
1         73.6    99.8   93.4   53.3   56.6   70.9  66.9  2005  
2         73.4    41.1   72.2   88.5   70.9   72.3  65.0  2005  
3         72.8    71.8   76.0   69.4   73.9   72.2  52.7  2005  
4         70.1    74.0   80.6   66.7   65.8   64.3  53.0  2005

In [14]:
data5 = pd.read_csv("./kaggle_université/timesData.csv", encoding='latin-1', sep=",")
df5 = pd.DataFrame(data5)
df5.head()

world_rank                        university_name                   country  \
0          1                     Harvard University  United States of America   
1          2     California Institute of Technology  United States of America   
2          3  Massachusetts Institute of Technology  United States of America   
3          4                    Stanford University  United States of America   
4          5                   Princeton University  United States of America   

   teaching international  research  citations income total_score  \
0      99.7          72.4      98.7       98.8   34.5        96.1   
1      97.7          54.6      98.0       99.9   83.7        96.0   
2      97.8          82.3      91.4       99.9   87.5        95.6   
3      98.3          29.5      98.1       99.2   64.3        94.3   
4      90.9          70.3      95.4       99.9      -        94.2   

  num_students  student_staff_ratio international_students female_male_ratio  \
0       20,152                  8.9                    25%               NaN   
1        2,243                  6.9                    27%           33 : 67   
2       11,074                  9.0                    33%           37 : 63   
3       15,596                  7.8                    22%           42 : 58   
4        7,929                  8.4                    27%           45 : 55   

   year  
0  2011  
1  2011  
2  2011  
3  2011  
4  2011

In [15]:
data6 = pd.read_csv("./kaggle_université/shanghaiData.csv", encoding='latin-1', sep=",", engine = "python")
df6 = pd.DataFrame(data6)
df6.head()

world_rank                              university_name national_rank  \
0          1                           Harvard University             1   
1          2                      University of Cambridge             1   
2          3                          Stanford University             2   
3          4           University of California, Berkeley             3   
4          5  Massachusetts Institute of Technology (MIT)             4   

   total_score  alumni  award   hici     ns    pub   pcp  year  
0        100.0   100.0  100.0  100.0  100.0  100.0  72.4  2005  
1         73.6    99.8   93.4   53.3   56.6   70.9  66.9  2005  
2         73.4    41.1   72.2   88.5   70.9   72.3  65.0  2005  
3         72.8    71.8   76.0   69.4   73.9   72.2  52.7  2005  
4         70.1    74.0   80.6   66.7   65.8   64.3  53.0  2005

### Question 2 : Quelles années sont prises en charge par chaque classement ?

In [130]:
df1.year.unique()

array([2012, 2013, 2014, 2015])

In [131]:
df4.year.unique()

array([2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015])

In [132]:
df5.year.unique()

array([2011, 2012, 2013, 2014, 2015, 2016])

In [133]:
df6.year.unique()

array([2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015])

### Réponse : df1 de 2012 à 2015, df2 de 1985 à 2015, df3 pas concerné, df4 de 2005 à 2015, df5 de 2011 à 2016 et df6 de 2005 à 2015

### Question : Combien d'universités sont prises en compte annuellement par chaque classement en moyenne ? Il s'agit ici de donner le nombre moyen d'universités prises en compte par chaque classement lors d'une année.

In [38]:
print(
    'DF1 :', df1['year'].value_counts().mean(),
    'DF4 :', df4['year'].value_counts().mean(),
    'DF5 :', df5['year'].value_counts().mean(), 
    'DF6 :', df6['year'].value_counts().mean()
)

DF1 : 550.0 DF4 : 445.1818181818182 DF5 : 433.8333333333333 DF6 : 445.1818181818182


### Réponse DF1 : 550 DF4 : 445.2 DF5 : 433.8 DF6 : 445.2

###   Question : Quels sont les rankings obtenus par l'Université de Strabourg dans chacun des 3 classements disponibles ?


In [66]:
df_strasbourg = pd.concat([df5, df4, df6])
cond_ = (df_strasbourg["university_name"].str.contains("University of Strasbourg", na=False))
df_strasbourg.loc[cond_,:][['world_rank','university_name', 'year']]

world_rank           university_name  year
422     201-225  University of Strasbourg  2012
825     201-225  University of Strasbourg  2013
1223    201-225  University of Strasbourg  2014
1619    201-225  University of Strasbourg  2015
2142    301-350  University of Strasbourg  2016
91           92  University of Strasbourg  2005
595          96  University of Strasbourg  2006
1100         99  University of Strasbourg  2007
1656    101-151  University of Strasbourg  2008
2158    101-151  University of Strasbourg  2009
2657    101-150  University of Strasbourg  2010
3157    102-150  University of Strasbourg  2011
3659    101-150  University of Strasbourg  2012
3894         97  University of Strasbourg  2013
3991         95  University of Strasbourg  2014
4485         87  University of Strasbourg  2015
91           92  University of Strasbourg  2005
595          96  University of Strasbourg  2006
1100         99  University of Strasbourg  2007
1656    101-151  University of Strasbourg  2008
2158    101-151  University of Strasbourg  2009
2657    101-150  University of Strasbourg  2010
3157    102-150  University of Strasbourg  2011
3659    101-150  University of Strasbourg  2012
3894         97  University of Strasbourg  2013
3991         95  University of Strasbourg  2014
4485         87  University of Strasbourg  2015

### Réponse : Voici

### Question : Selon le classement du Times, quels sont les 20 meilleures universités en 2016 ? Quelle est la part d'universités américaines ?


In [101]:
df_result = df5.loc[df5['year'] == 2016].head(20)

df_result

world_rank                                    university_name  \
1803          1                 California Institute of Technology   
1804          2                               University of Oxford   
1805          3                                Stanford University   
1806          4                            University of Cambridge   
1807          5              Massachusetts Institute of Technology   
1808          6                                 Harvard University   
1809          7                               Princeton University   
1810          8                            Imperial College London   
1811          9  ETH Zurich – Swiss Federal Institute of Tech...   
1812         10                              University of Chicago   
1813         11                           Johns Hopkins University   
1814         12                                    Yale University   
1815         13                 University of California, Berkeley   
1816         14                          University College London   
1817         15                                Columbia University   
1818         16              University of California, Los Angeles   
1819         17                         University of Pennsylvania   
1820         18                                 Cornell University   
1821         19                              University of Toronto   
1822         20                                    Duke University   

                       country  teaching international  research  citations  \
1803  United States of America      95.6          64.0      97.6       99.8   
1804            United Kingdom      86.5          94.4      98.9       98.8   
1805  United States of America      92.5          76.3      96.2       99.9   
1806            United Kingdom      88.2          91.5      96.7       97.0   
1807  United States of America      89.4          84.0      88.6       99.7   
1808  United States of America      83.6          77.2      99.0       99.8   
1809  United States of America      85.1          78.5      91.9       99.3   
1810            United Kingdom      83.3          96.0      88.5       96.7   
1811               Switzerland      77.0          97.9      95.0       91.1   
1812  United States of America      85.7          65.0      88.9       99.2   
1813  United States of America      77.6          70.0      90.4       98.2   
1814  United States of America      86.5          64.3      87.8       97.2   
1815  United States of America      80.4          61.9      91.1       99.7   
1816            United Kingdom      78.1          94.4      91.0       94.2   
1817  United States of America      85.9          73.5      82.2       98.1   
1818  United States of America      80.8          56.4      88.6       98.5   
1819  United States of America      82.0          49.5      86.9       98.6   
1820  United States of America      77.9          63.9      86.1       97.2   
1821                    Canada      75.9          77.8      89.3       90.9   
1822  United States of America      76.0          56.5      78.0       99.0   

     income total_score num_students  student_staff_ratio  \
1803   97.8        95.2        2,243                  6.9   
1804   73.1        94.2       19,919                 11.6   
1805   63.3        93.9       15,596                  7.8   
1806   55.0        92.8       18,812                 11.8   
1807   95.4        92.0       11,074                  9.0   
1808   45.2        91.6       20,152                  8.9   
1809   52.1        90.1        7,929                  8.4   
1810   53.7        89.1       15,060                 11.7   
1811   80.0        88.3       18,178                 14.7   
1812   36.6        87.9       14,221                  6.9   
1813  100.0        87.6       15,128                  3.6   
1814   43.3        87.4       11,751                  4.4   
1815   47.9        87.2       36,186                 16.4   
1816   40.5        8

In [98]:
(df_result['country'] == "United States of America").value_counts()


True     14
False     6
Name: country, dtype: int64

### Réponse : il y a le tableau et 14 universités Américaines

### Question : Combien chaque pays possède d'universités dans le classement de Shanghai en 2015 ? Range la liste par ordre décroissant. Pense à t'aider de la table université - pays pour réaliser cette requête.

In [ ]:
df_merged = pd.merge(df6, df3, left_on="university_name", right_on="school_name")
df_merged = df_merged.loc[df_merged['year'] == 2015]

df_merged['country'].value_counts().sort_values(ascending=True)


### Réponse : au dessus 

### Question : Quelles sont les universités françaises présentes dans le classement du Center for WUR ?


In [128]:
df1.loc[df1["country"] == 'France'].institution.unique()

array(['University of Paris-Sud', 'Ã\x89cole normale supérieure - Paris',
       'Ã\x89cole Polytechnique', 'Pierre-and-Marie-Curie University',
       'Mines ParisTech', 'Paris Diderot University - Paris 7',
       'University of Strasbourg', 'Joseph Fourier University',
       'Aix-Marseille University', 'Claude Bernard University Lyon 1',
       'University of Montpellier 2',
       'Paul Sabatier University - Toulouse III',
       'Paris Descartes University', 'University of Lorraine',
       'University of Bordeaux II',
       'Versailles Saint-Quentin-en-Yvelines University',
       'University of Bordeaux I',
       'Ã\x89cole normale supérieure de Lyon', 'University of Rennes 1',
       'Grenoble Institute of Technology', 'Ã\x89cole Centrale Paris',
       'University of Montpellier 1',
       'University of Nice Sophia Antipolis', 'University of Savoy',
       'University of Nantes',
       'University Lille 1: Sciences and Technologies',
       'National Polytechnic Institu

### Réponse : voici

### Question : Quels pays ont les plus faibles ratios (Dépenses dans les Institutions du Supérieur (Privées et Publiques) en 2011) / Nombre d'universités dans le classement du Times en 2011 ? Range cette liste par ordre croissant.


### Pas compris où sont les dépenses 